tests/src/Unit/OgPermissionHandlerTest.php

```diff
@@ -0,0 +1,214 @@
+    return $extension;
+  }
+
+  /**
+   * Tests permissions provided by YML files.
+   *
+   * @covers ::__construct
+   * @covers ::getPermissions
+   * @covers ::buildPermissionsYaml
+   * @covers ::moduleProvidesPermissions
+   */
+  public function testBuildPermissionsYaml() {
+    // defining OG_ANONYMOUS_ROLE on the fly since OG isn't enabled for this
+    // test and that could cause for a test's failure.
+    define('OG_ANONYMOUS_ROLE', 'non-member');
+
+    vfsStreamWrapper::register();
+    $root = new vfsStreamDirectory('modules');
+    vfsStreamWrapper::setRoot($root);
+
+    $this->moduleHandler = $this->getMock('Drupal\Core\Extension\ModuleHandlerInterface');
+    $this->moduleHandler->expects($this->once())
+      ->method('getModuleDirectories')
+      ->willReturn(array(
+        'module_a' => vfsStream::url('modules/module_a'),
+        'module_b' => vfsStream::url('modules/module_b'),
+      ));
+
+    $url = vfsStream::url('modules');
+    mkdir($url . '/module_a');
+    file_put_contents($url . '/module_a/module_a.og_permissions.yml',
+      "access_module_a: single_description"
+    );
+    mkdir($url . '/module_b');
+    file_put_contents($url . '/module_b/module_b.og_permissions.yml',
+      "'access module b':
+  title: 'Access B'
+  description: 'bla bla'
+");
+
+    $modules = array('module_a', 'module_b');
+    $extensions = array(
+      'module_a' => $this->mockModuleExtension('module_a', 'Module a'),
+      'module_b' => $this->mockModuleExtension('module_b', 'Module b'),
+    );
+
+    $this->moduleHandler->expects($this->any())
+      ->method('getModuleList')
+      ->willReturn(array_flip($modules));
+
+    $this->controllerResolver->expects($this->never())
+      ->method('getControllerFromDefinition');
+
+    $this->permissionHandler = new TestPermissionHandler($this->moduleHandler, $this->stringTranslation, $this->controllerResolver);
+
+    // Setup system_rebuild_module_data().
+    $this->permissionHandler->setSystemRebuildModuleData($extensions);
+
+    $actual_permissions = $this->permissionHandler->getPermissions();
+    $this->assertPermissions($actual_permissions);
+
+    $this->assertTrue($this->permissionHandler->moduleProvidesPermissions('module_a'));
+    $this->assertTrue($this->permissionHandler->moduleProvidesPermissions('module_b'));
+    $this->assertFalse($this->permissionHandler->moduleProvidesPermissions('module_c'));
+  }
+
+  /**
+   * Checks that the permissions are like expected.
+   *
+   * @param array $actual_permissions
+   *   The actual permissions
+   */
+  protected function assertPermissions(array $actual_permissions) {
+    $this->assertCount(2, $actual_permissions);
+
+    $this->assertEquals($actual_permissions['access_module_a']['title'], 'single_description');
+    $this->assertEquals($actual_permissions['access_module_a']['provider'], 'module_a');
+    $this->assertEquals($actual_permissions['access_module_a']['default role'], [OG_ANONYMOUS_ROLE]);
```

Comment:
Ideally we should not be using module file defined  global constant in unit tests as e.g. running this test standalone will likely break.

Same as above, we can follow up this stuff if needed.